### ISSUE TITLE
Events occasionally logged as sent but not shown in Slack

### ISSUE THREAD

**Issue by ejamesc** [jamescarlos/slackcamp#3]
I can't pinpoint this yet, but I've observed some Basecamp events not appearing in Slack even though it's been logged as sent from Slackcamp. This happens sporadically - sometimes it doesn't show up, other times it does. Have you experienced similar things?

(I'll try and track it down when I have more time).

**Comment by jamescarlos** [comment]
Hey @ejamesc. Yes, unfortunately I have also experienced this issue and have not been able to identify the issue yet. If you do find something, please don't hesitate to contribute.

**Comment by ejamesc** [comment]
Thanks for the prompt reply, James. I was worried I was doing something wrong (or imagining things!) 

I've just set up Slack for my company, so I'm hopeful that I'll find something in the current period, where I'm still fiddling with our internal server setup.

**Comment by dafeder** [comment]
Have had the same issue... seems to come and go? Weird...

**Comment by floodlightdesign** [comment]
Having the same issue except I've never seen the messages show up successfully in Slack. Cron job output seems to indicate that the messages are sent correctly. Any thoughts?

**Comment by jamescarlos** [comment]
@dafeder @floodlightdesign yeah it's odd. The script is quite simple, but some communication is getting lost along the way. I'm going to give another go at it soon, but let me know if you guys have any ideas.

**Comment by floodlightdesign** [comment]
@jamescarlos your code looks solid...nothing irregular. We'll try to do some debugging this weekend.

**Comment by eabruzzese** [comment]
After some debugging, it looks like Slack is returning a `Payload is not valid JSON` response for some messages.

Here's the JSON that I'm using (from a `json_encode($payload)`):

``` json
{
    "channel": "#basecamp",
    "username": "basecamp",
    "text": "Eric Abruzzese commented on <https://basecamp.com/1837925/projects/4271596/messages/33372683#comment_216698052|Notification Testing Procedure>",
    "icon_url": "http://i.imgur.com/yXujl8r.png",
    "attachments": [
        {
            "fallback": "Hmm, it&#39;s possible. Can you look at the spreadsheet posted near the top of the thread? There&#39;s a list of users affected by the exit surve...",
            "fields": [
                {
                    "title": "Eric Abruzzese",
                    "value": "Hmm, it&#39;s possible. Can you look at the spreadsheet posted near the top of the thread? There&#39;s a list of users affected by the exit surve...",
                    "short": false
                }
            ]
        }
    ]
}
```

It's obviously valid, so I'm not sure what's setting it off. Ideas? It happens consistently with this message.

Eric

**Comment by jamescarlos** [comment]
Thanks @eabruzzese for the fix: cd9f3ee268e9a5a8df45cb170eb210934b59b04d. Closing this out for now.
